server/service/src/plugin_files.rs

```diff
@@ -0,0 +1,106 @@
+use std::{fs, io::Error, path::PathBuf, str::FromStr};
+
+#[derive(Debug, PartialEq)]
+pub struct PluginFile {
+    pub config: String,
+    pub name: String,
+    pub path: String,
+}
+
+const PLUGIN_FILE_DIR: &'static str = "plugins";
+const PLUGIN_FILE_NAME: &'static str = "plugin.json";
+
+pub struct PluginFileService {
+    pub dir: PathBuf,
+}
+
+impl PluginFileService {
+    pub fn find_file(
+        base_dir: &Option<String>,
+        plugin: &str,
+        filename: Option<&str>,
+    ) -> anyhow::Result<Option<PluginFile>> {
+        let (plugin_dir_path, _plugin_dir) = get_plugin_dir(base_dir)?;
+
+        let filename = match filename {
+            Some(filename) => filename.to_string(),
+            None => format!("{}.js", plugin).clone(),
+        };
+        let file_path = match find_file(&plugin_dir_path, plugin, &filename)? {
+            Some(path) => path,
+            None => return Ok(None),
+        };
+        Ok(Some(PluginFile {
+            name: plugin.to_string(),
+            path: file_path.to_string_lossy().to_string(),
+            config: fs::read_to_string(plugin_dir_path.join(plugin).join(PLUGIN_FILE_NAME))?,
```

Comment:
I would imagine part of config would be parsable (and standard), to hold the `type` and version, probably not needed for prototype